packages/vite/src/node/optimizer/index.ts

```diff
@@ -1182,8 +1182,14 @@ const lockfileFormats = [
     manager: 'yarn',
   },
   {
-    // Yarn PnP
-    path: '.yarn/install-state',
+    // Yarn v3+ PnP
+    path: '.pnp.cjs',
+    checkPatches: false,
```

Comment:
I noticed that with `.pnp.cjs` doesn't contain information about patches anymore, so we likely need to re-enable `checkPatches` here. Since it's `.yarn/patches` instead of just `patches`, maybe we can modify this field to allow specifying the directory path instead of only a boolean.

Reply:
@smeng9 this isn't addressed. `checkPatches: true` checks `<root>/patches`. We want to check `<root>/.yarn/patches`

Reply:
the .yarn folder is added